File: Frontend/loan-management-frontend/src/app/models/loan-type.ts
export interface LoanType {
  loanTypeId: number;
  loanTypeName: string;
  interestRate: number;
  minAmount: number;
  maxAmount: number;
  maxTenureMonths: number;
  hasMoratorium: boolean;
  moratoriumMonths?: number;
  isActive: boolean;
  createdBy: string;
  createdOn: string;
  updatedBy?: string;
  updatedOn?: string;
}

File: Backend/LoanManagementSystem.Api/Models/LoanType.cs
using System.ComponentModel.DataAnnotations;

namespace LoanManagementSystem.Api.Models
{
    public class LoanType
    {
        [Key]
        public int LoanTypeId { get; set; }

        [Required, MaxLength(50)]
        public string LoanTypeName { get; set; } = string.Empty;

        [Required]
        public decimal InterestRate { get; set; } // Fixed interest

        [Required]
        public decimal MinAmount { get; set; }

        [Required]
        public decimal MaxAmount { get; set; }

        [Required]
        public int MaxTenureMonths { get; set; }

        public bool IsActive { get; set; } = true;
        public int? MoratoriumMonths { get; set; } // Only for education loan
        public bool HasMoratorium { get; set; }
        public int? UpdatedBy { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public int? CreatedBy { get; set; }
        public DateTime? CreatedOn { get; set; }



        // Navigation
        public ICollection<LoanApplication> LoanApplications { get; set; } = new List<LoanApplication>();
    }
}
